web/src/components/storage/ProposalSpacePolicySection.jsx

```diff
@@ -0,0 +1,457 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect } from "react";
+import { FormSelect, FormSelectOption } from "@patternfly/react-core";
+
+import { _, n_, N_ } from "~/i18n";
+import { deviceSize } from '~/components/storage/utils';
+import { If, OptionsPicker, Section, SectionSkeleton } from "~/components/core";
+import { noop, useLocalStorage } from "~/utils";
+import { sprintf } from "sprintf-js";
+import { Table, Thead, Tr, Th, Tbody, Td, TreeRowWrapper } from '@patternfly/react-table';
+
+/**
+ * @typedef {import ("~/client/storage").ProposalManager.ProposalSettings} ProposalSettings
+ * @typedef {import ("~/client/storage").ProposalManager.SpaceAction} SpaceAction
+ * @typedef {import ("~/client/storage").DevicesManager.StorageDevice} StorageDevice
+ */
+
+/**
+ * @typedef SpacePolicy
+ * @type {object}
+ * @property {string} name
+ * @property {string} label
+ * @property {string} description
+ */
+
+/** @type {SpacePolicy[]} */
+const SPACE_POLICIES = [
+  {
+    name: "delete",
+    label: N_("Delete current content"),
+    description: N_("All partitions will be removed and any data in the disks will be lost.")
+  },
+  {
+    name: "resize",
+    label: N_("Shrink existing partitions"),
+    description: N_("The data is kept, but the current partitions will be resized as needed.")
+  },
+  {
+    name: "keep",
+    label: N_("Use available space"),
+    description: N_("Existing partitions and data will not be modified. Only the space not assigned to any partition will be used.")
+  },
+  {
+    name: "custom",
+    label: N_("Custom"),
+    description: N_("Select what to do with each partition.")
+  }
+];
+
+// Names of the columns for the policy actions.
+const columnNames = {
+  device: N_("Used device"),
+  content: N_("Current content"),
+  size: N_("Size"),
+  action: N_("Action")
+};
+
+/**
+ * Indicates whether a device is a drive (disk, RAID).
+ * @function
+ *
+ * @param {StorageDevice} device
+ * @returns {boolean}
+ */
+const isDrive = (device) => Object.keys(device).includes("vendor");
```

Comment:
NP: I think this should be something given by the backend. But, in any case, I'd move this helper funtion to storage-utils.

Reply:
The backend already gives that. I mean, all the drives implement the `Drive` D-Bus interface. Maybe we could add the `isDrive` attribute to the `StorageDevice` objects created by the storage js client. I would rethink this again when we start exporting the LVM volume groups.